spec/models/doorkeeper/access_token_spec.rb

```diff
@@ -143,6 +143,12 @@ module NoGenerate
         subject.resource_owner_id = nil
         expect(subject).to be_valid
       end
+
+      it 'is valid without resource_owner_id' do
+        # For resource owner credentials flow
+        subject.application_id = nil
+        expect(subject).to be_valid
+      end
```

Comment:
FYI, will need to test against pre-4.0.0rc1. I'm checking it at the moment and will follow-up.